paraatm/simulation_method/integrated_flight_health_monitoring.py

```diff
@@ -0,0 +1,302 @@
+"""
+NASA University Leadership Initiative program
+Information fusion for real-time national air transportation system prognostics under uncertainty
+
+    Integrated Flight Heatlrh Monitoring (IFHM) module
+    - Simulate normal flight operation
+    - Simulate flight with upset flight scenarios using 1) upgen sub module
+    - Monitor flight health in real-time and detect aircraft upsets
+      using 2) fhm sub module
+    - Generate output files to perform analysis
+
+    @author: Hyunseong Lee, Adaptive Intelligent Materials & Systems (AIMS) Center,
+             Arizona State University
+    
+    Last modified on 5/25/2020
+"""
+
+import time
+import os
+from jpype import *
+from array import *
+from shutil import copyfile
+from scipy import signal
+import numpy as np
+import math
+import pandas as pd
+import tensorflow as tf
+from tensorflow.keras import backend
+from tensorflow.keras.models import Model, load_model
+from tensorflow.keras.layers import Input, Dense, BatchNormalization
+from tensorflow.keras.regularizers import l1
+from tensorflow.keras import optimizers
+from paraatm.io.gnats import GnatsSimulationWrapper
+from paraatm.simulation_method.upset_gen import upgen
+from paraatm.simulation_method.flight_health_monitor import fhm
+
+class IFHM(GnatsSimulationWrapper):
+    def __init__(self, sim_inputs):
+        
+        # GNATS simulation input files
+        self.fp_file = sim_inputs["fp_file"]
+        self.mfl_file = sim_inputs["mfl_file"]
+        self.cruz_alt = sim_inputs["cruz_alt"]
+        self.upset_case = sim_inputs["upset_case"]
+        self.upset_init = sim_inputs["upset_init"]
+        self.upset_dur = sim_inputs["upset_dur"]
+        self.alt_rate_coef = sim_inputs["alt_rate_coef"]
+        self.tas_rate_coef = sim_inputs["tas_rate_coef"]
+        self.course_rate_coef = sim_inputs["course_rate_coef"]
+        self.th_Mahal_dist = sim_inputs["th_Mahal_dist"]
+        
+        # Define GNATS-Server environment (not standalone) & perform simulation
+        # Set PARA-ATM & GNATS folders
+        # PARA_ATM_Home = os.environ.get('PARA_ATM_Home')
+        GNATS_Home = os.environ.get("GNATS_Home")
```

Comment:
The environment variable used by GNATS and para-atm is actually `GNATS_HOME`.  If you make use of `GnatsEnvironment`, you might not need to use this environment variable directly.